core/frontend/src/tile/map/MapLayerFormatRegistry.ts

```diff
@@ -144,18 +144,52 @@ export class MapLayerFormatRegistry {
   public setAccessClient(formatId: string, accessClient: MapLayerAccessClient): boolean {
     const entry = this._formats.get(formatId);
     if (entry !== undefined) {
-      entry.accessClient = accessClient;
+      entry.client = accessClient;
       return true;
     }
     return false;
   }
 
-  /** @beta */
+  /**
+  * Returns the active session client for a given format.
+   *  @beta */
   public getAccessClient(formatId: string): MapLayerAccessClient | undefined {
     if (formatId.length === 0)
       return undefined;
 
-    return this._formats.get(formatId)?.accessClient;
+    const accessClient = this._formats.get(formatId)?.client;
+    if (accessClient && typeof((accessClient as any).getAccessToken) === "function") {
+      return accessClient as MapLayerAccessClient;
+    }
+    return undefined;
+  }
+
+  /**
+  * Returns the active session client for a given format.
+  * @beta
+  */
+  public getSessionClient(formatId: string): MapLayerSessionClient | undefined {
+    if (formatId.length === 0)
+      return undefined;
+
+    const client = this._formats.get(formatId)?.client;
+    if (client && typeof((client as any).getSessionManager) === "function") {
```

Comment:
Same as previous comment, when will `getSessionManager` not be a function?

Reply:
I could, but why not be more restrictive and also check its a function ?

Reply:
Because you know it's a function (or undefined), right? Aren't we the ones setting this? We can trust our types within our own code boundary.

Reply:
Alright, changed it to check `undefined` instead.